workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/DialogStack.kt

```diff
@@ -0,0 +1,104 @@
+package com.squareup.workflow1.ui.container
+
+import android.content.Context
+import android.os.Parcel
+import android.os.Parcelable
+import android.os.Parcelable.Creator
+import android.view.View
+import androidx.lifecycle.LifecycleOwner
+import com.squareup.workflow1.ui.ViewEnvironment
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.androidx.WorkflowLifecycleOwner
+import com.squareup.workflow1.ui.container.DialogRunner.KeyAndBundle
+
+/**
+ * Does the work of maintaining a set of [Dialog][android.app.Dialog]s
+ * to reflect lists of [Overlay].
+ */
+@WorkflowUiExperimentalApi
+public class DialogStack(
+  private val context: Context,
+  private val getParentLifecycleOwner: () -> LifecycleOwner?,
+  private val createRectProvider: (Overlay) -> RectProvider
```

Comment:
`createRectProvider` could use some kdoc explaining why we might want to override it.

Reply:
Yeah. But I've just ripped it out of this PR, and I suspect it will come back in a different form.